src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -71,7 +71,8 @@ public PersonCard(Person person, int displayedIndex) {
             goalText = "No goal set for this person";
         } else {
             LocalDate deadline = person.getGoalDeadline(LocalDate.now());
-            if (deadline.getYear() == DateUtil.ZERO_DAY.getYear()) {
+            LocalDate zeroDay = DateUtil.ZERO_DAY;
+            if (deadline.equals(zeroDay)) {
```

Comment:
Probably no need for a new variable here. Perhaps just `deadline.equals(DateUtil.ZERO_DAY)`?